packages/botonic-plugin-contentful/src/contentful/manage/manage-entry.ts

```diff
@@ -0,0 +1,176 @@
+// eslint-disable-next-line node/no-missing-import
+import { ClientAPI } from 'contentful-management/dist/typings/create-contentful-api'
+// eslint-disable-next-line node/no-missing-import
+import { Entry } from 'contentful-management/dist/typings/entities/entry'
+// eslint-disable-next-line node/no-missing-import
+import { Environment } from 'contentful-management/dist/typings/entities/environment'
+
+import { CmsException, ContentId, ContentType } from '../../cms'
+import { ResourceNotFoundCmsException } from '../../cms/exceptions'
+import {
+  CONTENT_FIELDS,
+  ContentField,
+  ContentFieldType,
+} from '../../manage-cms/fields'
+import { ManageContext } from '../../manage-cms/manage-context'
+import * as nlp from '../../nlp'
+import { ContentfulOptions } from '../../plugin'
+import { isOfType } from '../../util/enums'
+
+export class ManageContentfulEntry {
+  constructor(
+    readonly options: ContentfulOptions,
+    readonly manage: ClientAPI,
+    readonly environment: Promise<Environment>
+  ) {}
+
+  async updateFields(
+    context: ManageContext,
+    contentId: ContentId,
+    fields: { [contentFieldType: string]: any }
+  ): Promise<{ [contentFieldType: string]: any }> {
+    const environment = await this.environment
+    const oldEntry = await this.getEntry(environment, contentId)
+    let needUpdate = false
+    for (const key of Object.keys(fields)) {
+      if (!isOfType(key, ContentFieldType)) {
+        throw new CmsException(`'${key}' is not a valid content field type`)
+      }
+      const field = this.checkOverwrite(context, oldEntry, key, false)
+      if (oldEntry.fields[field.cmsName][context.locale] === fields[key]) {
+        continue
+      }
+      needUpdate = true
+      oldEntry.fields[field.cmsName][context.locale] = fields[key]
+    }
+    if (!needUpdate) {
+      return oldEntry.fields
+    }
+    // we could use this.deliver.contentFromEntry & IgnoreFallbackDecorator to convert
+    // the multilocale fields returned by update()
+    await this.writeEntry(context, oldEntry)
+    return oldEntry.fields
+  }
+
+  async createContent(
+    context: ManageContext,
+    model: ContentType,
+    id: string
+  ): Promise<void> {
+    const environment = await this.environment
+    try {
+      await environment.createEntryWithId(model, id, {
+        fields: {},
+      })
+    } catch (e) {
+      throw new CmsException('ERROR while creating content', e)
```

Comment:
add the id of failing resources? `ERROR while creating content with id: '${id}'`